### PR TITLE
docs: DEV-2941: Release Notes for LSE 2.3

### PR DESCRIPTION
**Pull request by anithamuthu2022**

### CODE REVIEW

**Comment by anithamuthu2022** [comment]
@makseq @niklub feel free to pull this branch down and make changes accordingly.

File: docs/source/guide/release_notes.md
Comment (makseq) [inline docs/source/guide/release_notes.md:37]: We don't need to publish [internal] things

File: docs/source/guide/release_notes.md
Comment (makseq) [inline docs/source/guide/release_notes.md:40]: I would group everything regarding comments into one group and start with `Introduce the [**Comments**](comments_annotations.html) section for annotation editor. ...`

**Review by makseq — CHANGES_REQUESTED** [review]

**Comment by anithamuthu2022** [comment]
@makseq I've addressed your feedback for attention. Thank you!

**Comment by anithamuthu2022** [comment]
@makseq could you confirm if **Notifications** related entries should be removed from this PR?
Reference: https://heartex.slack.com/archives/C03DQ8U76PP/p1659395055710989?thread_ts=1659394615.296519&cid=C03DQ8U76PP

**Comment by anithamuthu2022** [comment]
@makseq I've removed `Comments` and `Notifications` entries in LSE 2.3 section as per https://heartex.slack.com/archives/C03DQ8U76PP/p1659395055710989?thread_ts=1659394615.296519&cid=C03DQ8U76PP.

**Comment by anithamuthu2022** [comment]
@makseq could you review and approve this PR for today's release.

**Review by makseq — APPROVED** [review]

**Comment by makseq** [comment]
> @makseq could you confirm if **Notifications** related entries should be removed from this PR? Reference: https://heartex.slack.com/archives/C03DQ8U76PP/p1659395055710989?thread_ts=1659394615.296519&cid=C03DQ8U76PP

Yes, notifications are not implemented and postponed to the next releases

**Comment by anithamuthu2022** [comment]
@niklub @joealfaro could you please review and approve this PR? I got the information from Max to take your approvals before the PR merge.

**Review by joealfaro — APPROVED** [review]

File: docs/source/guide/release_notes.md
Comment (niklub) [inline docs/source/guide/release_notes.md:27]: Combine 27,28 into one line

File: docs/source/guide/release_notes.md
Comment (niklub) [inline docs/source/guide/release_notes.md:32]: remove this

File: docs/source/guide/release_notes.md
Comment (niklub) [inline docs/source/guide/release_notes.md:26]: is there a link to rich description here?

File: docs/source/guide/release_notes.md
Comment (niklub) [inline docs/source/guide/release_notes.md:40]: I see this ticket and other ones refer to the Outliner feature that is not a part of this release. Please remove everything related to the Outliner Epic

File: docs/source/guide/release_notes.md
Comment (niklub) [inline docs/source/guide/release_notes.md:64]: Group all security fixes into a separate section

**Review by niklub — CHANGES_REQUESTED** [review]

File: docs/source/guide/release_notes.md
Comment (anithamuthu2022) [reply docs/source/guide/release_notes.md:26]: No.

File: docs/source/guide/release_notes.md
Comment (anithamuthu2022) [reply docs/source/guide/release_notes.md:40]: I deleted entries based on https://heartex.atlassian.net/browse/DEV-1544. @niklub please let me know if there are any further changes in this regard.

**Pull request closed without merging**
